internal/entities/properties/properties.go

```diff
@@ -47,56 +52,60 @@ func (p *Property) AsBool() (bool, error) {
 	if p == nil {
 		return false, fmt.Errorf("property is nil")
 	}
-	return propertyValueAs[bool](p.value)
+	return propertyValueAs[bool](p.value.AsInterface())
 }
 
 // GetBool returns the boolean value, or false if the value is not a boolean
 func (p *Property) GetBool() bool {
-	val, err := p.AsBool()
-	if err != nil {
+	if p == nil {
 		return false
 	}
-	return val
+	return p.value.GetBoolValue()
 }
 
 // AsString returns the string value, or an error if the value is not a string
 func (p *Property) AsString() (string, error) {
 	if p == nil {
 		return "", fmt.Errorf("property is nil")
 	}
-	return propertyValueAs[string](p.value)
+	return propertyValueAs[string](p.value.AsInterface())
 }
 
 // GetString returns the string value, or an empty string if the value is not a string
 func (p *Property) GetString() string {
 	if p == nil {
 		return ""
 	}
-	val, err := p.AsString()
-	if err != nil {
-		return ""
-	}
-	return val
+	return p.value.GetStringValue()
 }
 
 // AsInt64 returns the int64 value, or an error if the value is not an int64
 func (p *Property) AsInt64() (int64, error) {
 	if p == nil {
 		return 0, fmt.Errorf("property is nil")
 	}
-	return propertyValueAs[int64](p.value)
+	fval, err := propertyValueAs[float64](p.value.AsInterface())
+	if err != nil {
+		return 0, fmt.Errorf("value is not of type int64: %w", err)
+	}
+	return int64(fval), nil
```

Comment:
This need more work because internally `structpb.Value` stores any numerical value as `float64`. In order to handle `int64` and `uint64` I will, when creating a property or properties, check if the leaf value is either of those and change the value to be an object containing our own type annotation and the stringified value. The reverse on getting the value.